rest-api/controllers/pokemon_feed.go

```diff
@@ -102,6 +106,81 @@ func Api_feed(w http.ResponseWriter, r *http.Request) {
 		json.NewEncoder(w).Encode(fs)
 	}
 }
+func worker_add(id int, jobs <-chan Pokemon, results chan<- Pokemon) {
+	for j := range jobs {
+		if total > 0 && total_per_worker[id-1] > 0 {
+			fmt.Println("worker", id, "started  job", j)
+			fmt.Println("worker", id, "finished job", j)
+			results <- j
+		}
+		total--
+		total_per_worker[id-1] = total_per_worker[id-1] - 1
+	}
+}
+func Worker(w http.ResponseWriter, r *http.Request) {
+	jobs := make(chan Pokemon, 100)
+	results := make(chan Pokemon, 100)
+
+	items_per_workers := r.URL.Query().Get("items_per_workers")
+	intValue_per_worker, _ := strconv.Atoi(items_per_workers)
+
+	type_value := r.URL.Query().Get("type") //odd or even
+
+	items := r.URL.Query().Get("items")
+
+	items_value, _ := strconv.Atoi(items)
+
+	total = items_value
+	fmt.Printf(items_per_workers)
+	for w := 1; w <= 3; w++ {
+		go worker_add(w, jobs, results)
+		total_per_worker[w-1] = intValue_per_worker
+	}
+	r.ParseMultipartForm(10 << 20)
```

Comment:
I know why this value works, but would be more meaningful a constant with a name like 'multiformSize' or so.
Also don't forget about error handling https://pkg.go.dev/net/http#Request.ParseMultipartForm

Reply:
thanks , it has been updated too in both things